hikari/impl/cache.py

```diff
@@ -1348,11 +1343,12 @@ def delete_voice_state(
         if not voice_state_data:
             return None
 
+        self._garbage_collect_member(guild_record, voice_state_data.member, decrement=1)
+
         if not guild_record.voice_states:
             guild_record.voice_states = None
+            self._remove_guild_record_if_empty(guild_id, guild_record)
 
```

Comment:
Why was his moved

Reply:
It removes an unnecessary call to `_remove_guild_record_if_empty` when not needed. Garbage collecting the member will have this exact check, so it just makes more sense to have it here instead, since only under this case it will make sense to call it